main.cpp

```diff
@@ -13,75 +13,654 @@ What you create in this project will be used as the basis of Project 5 in the co
+    // Material (plastic, metal) (std::string)
+    std::string material = "plastic";
+    // Caliber (float)
+    float caliber = 5.56f;
+    // Current amount of bullets left (int)
+    int currentCapacity = 30;
+
+    // Store rounds for pewpew
+    int holdRounds(int roundsLoaded, int roundsExpended); // Returns number of current rounds in capa
+    // Feed rounds into feeding system
+    int feedRounds(bool isChamberJammed);
+    // Prevent loading of incorrect rounds
+    void filterIncorrectRounds(bool isCaliberCorrect); 
+};
+/*
+Thing 8) Bullet
+5 properties:
+    1) Caliber (float)
+    2) Powder load (int)
+    3) Powder brand (std::string)
+    4) Bullet weight (grains) (float)
+    5) Bullet type (HP, AP, FMJ, HE) (std::string)
+3 things it can do:
+    1) Engage rifling in barrel
+    2) Accelerate when fired
+    3) Impact target downrange
+ */
+struct bullet
+{
+    // Caliber (float)
+    float caliber = 5.56f;
+    // Powder load (int)
+    int powderLoad = 10;
+    // Powder brand (std::string)
+    std::string powderBrand = "4198";
+    // Bullet weight (grains) (float)
+    float weight = 10.0;
+    // Bullet type (HP, AP, FMJ, HE) (std::string)
+    std::string type = "FMJ";
+
+    // Engage rifling in barrel
+    void engageRifling(int threading, int acceleration);
+    // Accelerate when fired
+    int accelerateWhenFired(int powderLoad, float caliber, float barrelLength); // Returns maximum speed of bullet at end of barrel
+    // Impact target downrange
+    int impactTarget(int distance, int elevation, int speed, int mass);// Returns predicted speed of bullet when hitting target downrange
+};
+/*
+Thing 9) Stock
+5 properties:
+    1) Is skeletonized (bool)
+    2) Is foldable (bool)
+    3) Has cheekrest (bool)
+    4) Has sling point (bool)
+    5) Weight (g) (float)
+3 things it can do:
+    1) Increase control of recoil in gun
+    2) Toggle folded/unfolded state
+    3) Provide support to firer's head when ADS
+ */
+struct stock
+{
+    // Is skeletonized (bool)
+    bool isSkeletonized = false;
+    // Is foldable (bool)
+    bool isFoldable = true;
+    // Has cheekrest (bool)
+    bool hasCheekrest = true;
+    // Has sling point (bool)
+    bool hasSlingPoint = true;
+    // Weight (g) (float)
+    float weight = 850.48f;
+
+    // Increase control of recoil in gun
+    int increaseControlRecoil(int recoilForce, float gunWeight); // Returns percentage of recoil control increase
+    // Toggle folded/unfolded state
+    void toggleFoldedUnfolded(bool isFoldable); //
+    // Provide support to firer's head when ADS
+    void supportHeadWhenADS(bool isAiming = true, float pressureAgainstStock = 0.0f);
+};
+/*
+Thing 10) Assault rifle
+5 properties:
+    1) Barrel
+    2) Scope
+    3) Magazine
+    4) Bullet
+    5) Stock
+3 things it can do: 
+    1) Discharge bullet
+    2) Chamber bullet
+    3) Eject empty magazine
+ */
+struct assaultRifle
+{ 
+    // Barrel
+    struct barrel
```

Comment:
You already have classes defined for the properties of the `AssaultRifle`. 
Declare instances of them inside this class.
Don't copy the definitions of those classes inside this class.
- [ ] revise